spyderlib/widgets/sourcecode/syntaxhighlighters.py

```diff
@@ -190,7 +195,35 @@ def set_color_scheme(self, color_scheme):
 
     def highlightBlock(self, text):
         raise NotImplementedError
-            
+
+    def highlight_spaces(self, text, offset=0):
+        """
+        Make blank space less apparent by setting the foreground alpha.
+        This only has an effect when 'Show blank space' is turned on.
+        Derived classes could call this function at the end of
+        highlightBlock().
+        """
+        show_blanks = self.document().defaultTextOption().flags() & QTextOption.ShowTabsAndSpaces
+        if show_blanks:
+            format_leading = self.formats.get("leading", None)
+            format_trailing = self.formats.get("trailing", None)
+            match = self.BLANKPROG.search(text, offset)
+            while match:
+                start, end = match.span()
+                start = max([0, start+offset])
+                end = max([0, end+offset])
+                # Format trailing spaces at the end of the line. 
+                if end == len(text) and format_trailing is not None:
+                    self.setFormat(start, end, format_trailing)
+                # Format leading spaces, e.g. indentation.
+                if start == 0 and format_leading is not None:
+                    self.setFormat(start, end, format_leading)
+                format = self.format(start)
+                color_foreground = format.foreground().color()
+                color_foreground.setAlphaF(self.BLANK_ALPHA_FACTOR * color_foreground.alphaF())

```

Comment:
This line is probably longer than 79?

I am actually flexible on that PEP8 rule but I think @ccordoba12 is more serious about it :-)